app/components/gh-file-uploader.js

```diff
@@ -150,10 +170,22 @@ export default Component.extend({
 
     actions: {
         fileSelected(fileList) {
-            this.set('file', fileList[0]);
-            run.schedule('actions', this, function () {
-                this.generateRequest();
-            });
+            // can't use array destructuring here as FileList is not a strict
+            // array and fails in Safari
+            // jscs:disable requireArrayDestructuring
+            let file = fileList[0];
+            // jscs:enable requireArrayDestructuring
+            let validationResult = invokeAction(this, 'validate', file);
```

Comment:
Instead of the hack above, couldn't you do something like:

```js
let validationResult = invokeAction(this, 'validate', file) || this._defaultValidate(file);
```

invokeAction will return undefined if no action exists for that name.